<launch>
  <arg name="device" default="/dev/video0"/>
  <arg name="gpio" default="51"/>
  <arg name="frame_id" default="camera0"/>
  <arg name="namespace" default="$(arg frame_id)"/>
  <arg name="phase" default="0"/>

  <arg name="frame_rate" default="10"/>
  <arg name="use_image_transport" default="false"/>
  <arg name="encoding" default="rgb8"/>
  <arg name="use_triggering" default="true"/>

  <node name="gscam_driver_v4l" pkg="gscam_x2" type="gscam_x2" output="screen" ns="$(arg namespace)">
    <param name="camera_name" value="$(arg frame_id)"/>
    <param name="frame_id" value="$(arg frame_id)"/>
    <param name="phase" value="$(arg phase)"/>
    <param name="gpio" value="$(arg gpio)"/>
    <param name="use_image_transport" value="$(arg use_image_transport)"/>
    <param name="use_triggering" value="$(arg use_triggering)"/>

    <param name="camera_info_url" value="package://gscam/examples/uncalibrated_parameters.ini"/>
    <param name="gscam_x2_config" value="v4l2src device=$(arg device) do-timestamp=true ! video/x-raw, width=1920, height=1280, framerate=30/1, format=UYVY"/>
    <param name="image_encoding" value="$(arg encoding)"/>
    <param name="sync_sink" value="true"/>
    <param name="frame_rate" value="$(arg frame_rate)"/>
  </node>

</launch>
